src/features/GitHubAuth/ui/GitHubAuth.tsx

```diff
@@ -0,0 +1,22 @@
+import { Button } from '@mantine/core';
+import { useUser } from 'entities/User';
+import { GithubAuthProvider, signInWithPopup } from 'firebase/auth'
+import { auth, githubProvider } from 'shared/firebase/firebase'
+
+export const GitHubAuth = () => {
+    const { setUser } = useUser();
+    const a = () => {
```

Comment:
```suggestion
    const a = () => {
```
dont use naming like this